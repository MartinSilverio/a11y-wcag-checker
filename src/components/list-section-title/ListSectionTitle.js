import React from 'react';
import './ListSectionTitle.scss';

function ListSectionTitle({ title }) {
  return <h2 className='list-section-title'>{title}</h2>;
}

export default ListSectionTitle;
