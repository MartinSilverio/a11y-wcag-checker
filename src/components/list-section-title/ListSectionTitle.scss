.list-section-title {
  font-size: 16px;
  padding-left: 10px;
}
