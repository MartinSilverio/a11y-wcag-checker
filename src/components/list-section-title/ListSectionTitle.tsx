import React from 'react';
import './ListSectionTitle.scss';

function ListSectionTitle({
  title
}: any) {
  // @ts-expect-error ts-migrate(17004) FIXME: Cannot use JSX unless the '--jsx' flag is provided... Remove this comment to see the full error message
  return <h2 className='list-section-title'>{title}</h2>;
}

export default ListSectionTitle;
