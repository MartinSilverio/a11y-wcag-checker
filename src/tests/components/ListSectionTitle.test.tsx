import React from 'react';
// @ts-expect-error ts-migrate(7016) FIXME: Could not find a declaration file for module 'enzy... Remove this comment to see the full error message
import { shallow } from 'enzyme';
// @ts-expect-error ts-migrate(6142) FIXME: Module '../../components/list-section-title/ListSe... Remove this comment to see the full error message
import ListSectionTitle from '../../components/list-section-title/ListSectionTitle';

describe('ListSectionTitle', () => {
  it('should have a title wrapped in h2 if title exists', () => {
    expect.assertions(1);

    // @ts-expect-error ts-migrate(17004) FIXME: Cannot use JSX unless the '--jsx' flag is provided... Remove this comment to see the full error message
    expect(shallow(<ListSectionTitle title='Hello World' />)).toMatchSnapshot();
  });

  it("should have closing h2 tag if title doesn't exist", () => {
    expect.assertions(1);

    // @ts-expect-error ts-migrate(17004) FIXME: Cannot use JSX unless the '--jsx' flag is provided... Remove this comment to see the full error message
    expect(shallow(<ListSectionTitle />)).toMatchSnapshot();
  });
});
