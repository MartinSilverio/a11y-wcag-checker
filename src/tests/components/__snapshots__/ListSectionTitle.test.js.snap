// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListSectionTitle should have a title wrapped in h2 if title exists 1`] = `
<h2
  className="list-section-title"
>
  Hello World
</h2>
`;

exports[`ListSectionTitle should have closing h2 tag if title doesn't exist 1`] = `
<h2
  className="list-section-title"
/>
`;
